### PR TITLE
Teacher's view of the courses and individual course views (+express routes)

### LINKED ISSUE #18: Task.15 - Create a home page that displays a teacher's courses once logged in
Create a page for teachers, once logged in, that displays a grid of courses that were previously created by the teacher. Each courses links to a separate page that will display the course code and all the registered participants in the class

### LINKED ISSUE #48: Task.20 - Create an express route to return all active courses a teacher has
Create an express route for the teacher's view

### PR DESCRIPTION
**Pull request by JenLys**
A page for the teacher to view all courses has been created as well as individual views for each course. Moreover, express routes were made to show all courses made by the teacher.
image

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -23,6 +24,7 @@ function App() {
           <Route path="/teacherreg" element={<TeacherRegistration />} />
           <Route path="/studentlogin" element={<StudentLogin />} />
           <Route path="/teacherlogin" element={<TeacherLogin />} />
+          <Route path="/teacherview" element={<TeacherView />} />
```
Comment (Marian-Hristov) [inline client/src/App.jsx:27]: please do not add a new page for this component, as I have mentionned before, not each component needs to have its own page. Component doesn't equal page

File: client/src/pages/TeacherView.jsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        const y = e.clientY - rect.top;
+        box.style.setProperty('--x', `${x}px`);
+        box.style.setProperty('--y', `${y}px`);
+      };
+    
+      const handleMouseLeave = (e) => {
+        const box = e.currentTarget;
+        box.style.setProperty('--x', `-9999px`);
+        box.style.setProperty('--y', `-9999px`);
+      };
```
Comment (Marian-Hristov) [inline client/src/pages/TeacherView.jsx]: I believe that here you are trying to conditionally render something when something is hovered on. Please do not use `getBoundingClientRect` or `setProperty` as the latter could lead to security issues.

What i suggest is use the `onHover` option of the component and that calls a function called `handleHover` that sets a boolean variable using useState and then that gets conditionally rendered.

File: client/src/pages/TeacherView.jsx
Comment (Marian-Hristov) [inline client/src/pages/TeacherView.jsx:1]: Correct me if I am wrong, but what I think you are trying to have too much functionality within a single component. Within a single component you are trying to render a list of courses, a single course view and a creating teams view. Divide those into each of those components that are accessed through the main component. For example, the parent component will display the buttons for navigating to the child component and will pass information to the child component meaning that the useEffect hook will be used in the parent component. Then if you have to render a component like TeacherTeamCreation, you will pass a course in its props (short for properties) and that will be accessed within that component. You should do the same for your order functionalities.

File: client/vite.config.js.timestamp-1729286432055-a60ea1fa06ca5.mjs
Comment (Marian-Hristov) [inline client/vite.config.js.timestamp-1729286432055-a60ea1fa06ca5.mjs:1]: @JenLys Please be careful with all the files that you push, you should always check before commit with `git status`

File: server/routes/teacherRouter.js [hunk 3]
```diff
@@ -18,6 +19,24 @@ teacherRouter.get("/login", async (req, res) => {
...
+    }
+
+    // Find courses by prof id
+    const courses = await Course.find({ prof_id: profId });
+    res.json(courses);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
+  }
+});
```
Comment (Marian-Hristov) [inline server/routes/teacherRouter.js]: A lot of validation is missing please referrer to the `/register` route

**Review by Marian-Hristov — COMMENTED** [review]
Overall great work, just a few more things to learn for React but that is normal! keep up the good work!

File: client/src/App.jsx (hunk 1 above)
Comment (JenLys) [reply client/src/App.jsx:27]: I see. I have made the teacher view a separate page and did separate renderings of each course page. Please let me know if you want me to include the teacher's view as a separate page where the courses will be rendered on the same page or if you wish the teacher view to get rendered from a different page (if so, which one). Thank you for your thorough review :))

File: client/src/pages/TeacherView.jsx (hunk 2 above)
Comment (JenLys) [reply client/src/pages/TeacherView.jsx]: Got it! Thank you for the feedback :)). I have fixed it and will commit my changes.

File: server/routes/teacherRouter.js (hunk 3 above)
Comment (JenLys) [reply server/routes/teacherRouter.js]: Thank you for the feedback! I have committed the changes

**Pull request closed without merging**